File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcRelationship/IfcRelDeclares/IfcRelDeclares.py
class IfcRelDeclares(IfcRelationship):
    def __init__(self):
        self.RelatingContext: IfcContext = None
        self.RelatedDefinitions: SET [1:?] OF IfcDefinitionSelect = None

    # === WHERE CLAUSES ===
    # NoSelfReference : SIZEOF(QUERY(Temp <* RelatedDefinitions | RelatingContext :=: Temp)) = 0

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcRelationship/IfcRelDeclares/IfcRelDeclares.cs
public class IfcRelDeclares : IfcRelationship
{
    public IfcContext RelatingContext { get; set; }
    public SET [1:?] OF IfcDefinitionSelect RelatedDefinitions { get; set; }

    // === WHERE CLAUSES ===
    // NoSelfReference : SIZEOF(QUERY(Temp <* RelatedDefinitions | RelatingContext :=: Temp)) = 0
}
